appsec/src/helper/remote_config/listeners/asm_features_listener.cpp

```diff
@@ -7,22 +7,31 @@
 #include "../../json_helper.hpp"
 #include "../../utils.hpp"
 #include "../exception.hpp"
-#include <algorithm>
+#include "config_aggregators/asm_features_aggregator.hpp"
 #include <rapidjson/document.h>
 
-void dds::remote_config::asm_features_listener::on_update(const config &config)
+namespace dds::remote_config {
+
+void asm_features_listener::init()
 {
-    rapidjson::Document serialized_doc;
+    ruleset_ = rapidjson::Document(rapidjson::kObjectType);
+    aggregator_ = std::make_unique<asm_features_aggregator>();
+    aggregator_->init(&ruleset_.GetAllocator());
+    service_config_->unset_asm();
+    service_config_->unset_auto_user_instrum();
```

Comment:
Thinking about this a bit more, perhaps it would be better to handle the absence of keys at the commit stage.

The reason is that this change may result in extensions temporarily disabled as the configurations are parsed.